### PR TITLE
new file:   molenc/molenc.16.5.0/opam

### PR DESCRIPTION
**Pull request by UnixJunkie**
New exe:
```bash
$ molenc_bbad
usage:
  molenc_bbad
  [-i <filename>]: SMILES input file
  -o <filename>: output file
  [-np <int>]: nprocs (default=1)
  [--bbad <fn1[,fn2[,...]]>]: previously computed BBAD
  to apply as filter
  will compute the union of ADs if several filenames
  [-v]: verbose/debug mode
```

An Atom-Pairs (AP) based Bounding Box Applicability Domain (BBAD)

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
The build fails with:
```
#=== ERROR while compiling molenc.16.5.0 ======================================#
# context              2.0.10 | linux/x86_64 | ocaml-base-compiler.4.07.1 | pinned(https://github.com/UnixJunkie/molenc/archive/v16.5.0.tar.gz)
# path                 ~/.opam/4.07/.opam-switch/build/molenc.16.5.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p molenc -j 31
# exit-code            1
# env-file             ~/.opam/log/molenc-14860-865888.env
# output-file          ~/.opam/log/molenc-14860-865888.out
### output ###
# File "src/dune", line 7, characters 40-44:
# 7 |   (libraries batteries dolog bst parany pyml))
#                                             ^^^^
# Error: Library "pyml" not found.
# Hint: try:
#   dune external-lib-deps --missing --no-config --root . --ignore-promoted-rules --default-target @install --always-show-command-line --promote-install-files --release --only-packages molenc -p molenc --profile release -j 31 @install
# File "src/dune", line 24, characters 46-50:
# 24 |              ocamlgraph vector3 line_oriented pyml))
#                                                    ^^^^
# Error: Library "pyml" not found.
# Hint: try:
#   dune external-lib-deps --missing --no-config --root . --ignore-promoted-rules --default-target @install --always-show-command-line --promote-install-files --release --only-packages molenc -p molenc --profile release -j 31 @install

- File "src/dune", line 7, characters 40-44:
- 7 |   (libraries batteries dolog bst parany pyml))
-                                             ^^^^
- Error: Library "pyml" not found.
- Hint: try:
-   dune external-lib-deps --missing --no-config --root . --ignore-promoted-rules --default-target @install --always-show-command-line --promote-install-files --release --only-packages molenc -p molenc --profile release -j 31 @install
- File "src/dune", line 24, characters 46-50:
- 24 |              ocamlgraph vector3 line_oriented pyml))
-                                                    ^^^^
- Error: Library "pyml" not found.
- Hint: try:
-   dune external-lib-deps --missing --no-config --root . --ignore-promoted-rules --default-target @install --always-show-command-line --promote-install-files --release --only-packages molenc -p molenc --profile release -j 31 @install
```

**Comment by UnixJunkie** [comment]
ok, pyml was added to the dune file but not in the opam file; I updated it

File: packages/molenc/molenc.16.5.0/opam [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  "bst" {>= "2.0.0"}
+  "conf-graphviz"
+  "conf-python-3"
+  "conf-rdkit"
+  "cpm" {>= "9.0.0"}
+  "dokeysto_camltc"
+  "dolog" {>= "5.0.0"}
+  "dune" {>= "1.11"}
+  "minicli" {>= "5.0.0"}
+  "ocaml" {>= "4.07"}
```
Comment (mseri) [inline packages/molenc/molenc.16.5.0/opam:36]:
```suggestion
  "ocaml" {>= "4.08"}
```
Uses List.filter_map

**Comment by mseri** [comment]
Thanks. Please upstream the lower bound on ocaml

**Comment by UnixJunkie** [comment]
done, thanks!

**Pull request merged**